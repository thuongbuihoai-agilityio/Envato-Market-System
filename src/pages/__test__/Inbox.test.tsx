import { render } from '@testing-library/react';

// Pages
import ChatMemberList from '@app/pages/Inbox';

jest.mock('@chakra-ui/react', () => {
  const actual = jest.requireActual('@chakra-ui/react');
  return {
    ...actual,
    useBreakpointValue: jest.fn(),
  };
});

describe('Chat Component', () => {
  it('Should render match with snapshot.', async () => {
    const { container } = render(<ChatMemberList />);

    expect(container).toMatchSnapshot();
  });
});
