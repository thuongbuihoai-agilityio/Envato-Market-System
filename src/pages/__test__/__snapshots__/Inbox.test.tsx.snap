// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Component Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-12z1u6b"
  >
    <div
      class="css-1nbk4bb"
    >
      <div
        class="css-1lekzkb"
      >
        <h3
          class="chakra-text css-1socs6x"
        >
          Messages
          <span
            class="chakra-text css-1socs6x"
          >
            (
            2
            )
          </span>
        </h3>
        <svg
          class="stroke-darkblack-400 dark:stroke-bgray-50"
          fill="none"
          height="25"
          stroke="#1A202C"
          viewBox="0 0 24 25"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 12.7508V18.7411C22 20.9466 20.2091 22.7346 18 22.7346H6C3.79086 22.7346 2 20.9466 2 18.7411V6.76059C2 4.55504 3.79086 2.76709 6 2.76709H12M15.6864 4.78655C15.6864 4.78655 15.6864 6.21452 17.1167 7.64249C18.547 9.07046 19.9773 9.07046 19.9773 9.07046M9.15467 16.7339L12.1583 16.3055C12.5916 16.2437 12.9931 16.0433 13.3025 15.7343L21.4076 7.64249C22.1975 6.85384 22.1975 5.57519 21.4076 4.78655L19.9773 3.35857C19.1873 2.56993 17.9066 2.56993 17.1167 3.35857L9.01164 11.4504C8.70217 11.7594 8.50142 12.1602 8.43952 12.5928L8.01044 15.5915C7.91508 16.2579 8.4872 16.8291 9.15467 16.7339Z"
            stroke-linecap="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
      <div
        class="css-3ymqf5"
      >
        <div
          class="css-16zaavb"
        >
          <div
            class="css-y72fik"
          >
            <div
              class="css-blvggq"
            >
              <span
                class="chakra-avatar css-100byw3"
              >
                <svg
                  aria-label=" avatar"
                  class="chakra-avatar__svg css-16ite8i"
                  role="img"
                  viewBox="0 0 128 128"
                >
                  <path
                    d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                    fill="currentColor"
                  />
                  <path
                    d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                    fill="currentColor"
                  />
                </svg>
                <div
                  class="chakra-avatar__badge css-1hero9b"
                />
              </span>
              <div
                class="css-1v994a0"
              >
                <p
                  class="chakra-text css-0"
                >
                  Kien Nguyen
                </p>
              </div>
            </div>
            <div
              class="css-j7qwjs"
            >
              <p
                class="chakra-text css-0"
              >
                16:46
              </p>
              <img
                alt="Attach icon"
                class="chakra-image__placeholder css-1clelm1"
                src="images/fallback.webp"
              />
            </div>
          </div>
        </div>
        <div
          class="css-16zaavb"
        >
          <div
            class="css-y72fik"
          >
            <div
              class="css-blvggq"
            >
              <span
                class="chakra-avatar css-100byw3"
              >
                <svg
                  aria-label=" avatar"
                  class="chakra-avatar__svg css-16ite8i"
                  role="img"
                  viewBox="0 0 128 128"
                >
                  <path
                    d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                    fill="currentColor"
                  />
                  <path
                    d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                    fill="currentColor"
                  />
                </svg>
                <div
                  class="chakra-avatar__badge css-aku9t4"
                />
              </span>
              <div
                class="css-1v994a0"
              >
                <p
                  class="chakra-text css-0"
                >
                  Huy Pham
                </p>
              </div>
            </div>
            <div
              class="css-j7qwjs"
            >
              <p
                class="chakra-text css-0"
              >
                16:46
              </p>
              <img
                alt="Attach icon"
                class="chakra-image__placeholder css-1clelm1"
                src="images/fallback.webp"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-15fvzy5"
    >
      <div
        class="css-g0ip2s"
      >
        <div
          class="css-1hoc0te"
        >
          <h3
            class="chakra-heading css-gh62sw"
          >
            <div
              class="css-16zaavb"
            >
              <div
                class="css-y72fik"
              >
                <div
                  class="css-blvggq"
                >
                  <span
                    class="chakra-avatar css-100byw3"
                  >
                    <svg
                      aria-label=" avatar"
                      class="chakra-avatar__svg css-16ite8i"
                      role="img"
                      viewBox="0 0 128 128"
                    >
                      <path
                        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                        fill="currentColor"
                      />
                      <path
                        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                        fill="currentColor"
                      />
                    </svg>
                    <div
                      class="chakra-avatar__badge css-1hero9b"
                    />
                  </span>
                  <div
                    class="css-1v994a0"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      undefined undefined
                    </p>
                  </div>
                </div>
                <div
                  class="css-j7qwjs"
                >
                  <p
                    class="chakra-text css-0"
                  />
                </div>
              </div>
            </div>
          </h3>
        </div>
        <div
          class="css-gpjsqf"
        >
          <div
            class="css-15rwwo"
          >
            <span
              class="chakra-avatar css-gyntuh"
            >
              <svg
                aria-label=" avatar"
                class="chakra-avatar__svg css-16ite8i"
                role="img"
                viewBox="0 0 128 128"
              >
                <path
                  d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                  fill="currentColor"
                />
                <path
                  d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                  fill="currentColor"
                />
              </svg>
            </span>
            <div
              class="css-cr18eb"
            >
              <div
                class="css-0"
                data-testid="image-container"
              >
                <p
                  class="chakra-text css-1nsc086"
                  data-testid="text-content"
                >
                  Hello, I want to know more about your product
                </p>
              </div>
              <div
                class="css-17xejub"
              />
              <p
                class="chakra-text css-haer8f"
              >
                3:51 PM
              </p>
            </div>
          </div>
          <div
            class="css-10zunq2"
          >
            <div
              class="css-e5lq4j"
            >
              <div
                class="css-0"
                data-testid="image-container"
              >
                <p
                  class="chakra-text css-ccwdrb"
                  data-testid="text-content"
                >
                  Hi, i am here for help
                </p>
              </div>
              <div
                class="css-17xejub"
              />
              <p
                class="chakra-text css-haer8f"
              >
                3:51 PM
              </p>
            </div>
            <span
              class="chakra-avatar css-ntspms"
              data-testid="avatar"
            >
              <svg
                aria-label=" avatar"
                class="chakra-avatar__svg css-16ite8i"
                role="img"
                viewBox="0 0 128 128"
              >
                <path
                  d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                  fill="currentColor"
                />
                <path
                  d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="css-fxiqdw"
          >
            <div
              data-testid="react-quill"
            />
            <button
              class="chakra-button css-mmaqpq"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.0586 7.09154L7.92522 3.52487C3.13355 1.12487 1.16689 3.09153 3.56689 7.8832L4.29189 9.3332C4.50022 9.7582 4.50022 10.2499 4.29189 10.6749L3.56689 12.1165C1.16689 16.9082 3.12522 18.8749 7.92522 16.4749L15.0586 12.9082C18.2586 11.3082 18.2586 8.69153 15.0586 7.09154ZM12.3669 10.6249H7.86689C7.52522 10.6249 7.24189 10.3415 7.24189 9.99987C7.24189 9.6582 7.52522 9.37487 7.86689 9.37487H12.3669C12.7086 9.37487 12.9919 9.6582 12.9919 9.99987C12.9919 10.3415 12.7086 10.6249 12.3669 10.6249Z"
                    fill="white"
                  />
                </svg>
              </span>
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
